spec/faraday/error_spec.rb

```diff
@@ -67,7 +67,45 @@
       end
 
       it 'allows backward-compatible class to be subclassed' do
-        expect { class CustomError < Faraday::Error::ClientError; end }.not_to raise_error
+        expect {
+          with_warn_squelching { Class.new(Faraday::Error::ClientError) }
+        }.not_to raise_error
+      end
+
+      it 'allows rescuing of a current error with a deprecated error' do
+        begin
+          raise Faraday::ClientError, nil
+        rescue Faraday::Error::ClientError
+        rescue Faraday::Error => exc
+          fail "rescued #{exc.class.name} instead"
+        end
+      end
+
+      it 'allows rescuing of a current error with a current error' do
+        begin
+          raise Faraday::ClientError, nil
+        rescue Faraday::ClientError
+        rescue Faraday::Error => exc
+          fail "rescued #{exc.class.name} instead"
+        end
+      end
+
+      it 'allows rescuing of a deprecated error with a deprecated error' do
+        begin
+          raise Faraday::Error::ClientError, nil
+        rescue Faraday::Error::ClientError
+        rescue Faraday::Error => exc
+          fail "rescued #{exc.class.name} instead"
+        end
+      end
+
+      it 'allows rescuing of a deprecated error with a current error' do
+        begin
+          raise Faraday::Error::ClientError, nil
+        rescue Faraday::ClientError
```

Comment:
Can we find a way to output a message of success when we pass here?

Reply:
What is a "not-fail" test case? Is that something you see in the results? What would a message of success here look like?

Reply:
I changed those examples to use `expect { raise ... }.to raise_error(...)` instead. Much shorter, and still tests the correct thing. If I remove the `#===` implementation on the inherited meta class that `Faraday::DeprecatedClass.proxy_class` creates, I get these failures:

```
  1) Faraday::DeprecatedClass allows rescuing of a current error with a deprecated error
     Failure/Error: expect { raise SampleClass, nil }.to raise_error(SampleDeprecatedClass)

       expected SampleDeprecatedClass, got #<SampleClass: SampleClass> with backtrace:
         # ./spec/faraday/deprecate_spec.rb:44:in `block (3 levels) in <top (required)>'
         # ./spec/faraday/deprecate_spec.rb:44:in `block (2 levels) in <top (required)>'
     # ./spec/faraday/deprecate_spec.rb:44:in `block (2 levels) in <top (required)>'

  2) Faraday::ClientError.initialize maintains backward-compatibility until 1.0 allows rescuing of a current error with a deprecated error
     Failure/Error: expect { raise Faraday::ClientError, nil }.to raise_error(Faraday::Error::ClientError)

       expected Faraday::Error::ClientError, got #<Faraday::ClientError Faraday::ClientError> with backtrace:
         # ./spec/faraday/error_spec.rb:76:in `block (5 levels) in <top (required)>'
         # ./spec/faraday/error_spec.rb:76:in `block (4 levels) in <top (required)>'
     # ./spec/faraday/error_spec.rb:76:in `block (4 levels) in <top (required)>'
```

Reply:
I was thinking of "if we  get here, `assert_equal(true, true, "it works")`" kind of thing.

Reply:
Ohh I see. I think the `expect` syntax works too. I don't think an `expect(true).to be_true` underneath is necessary.